cmd/tidb-lightning/main.go

```diff
@@ -74,11 +75,18 @@ func main() {
 		return
 	}
 
-	if cfg.App.ServerMode {
-		err = app.RunServer()
-	} else {
-		err = app.RunOnce()
-	}
+	err = func() error {
+		if globalCfg.App.ServerMode {
+			return app.RunServer()
+		} else {
+			cfg := config.NewConfig()
+			if err := cfg.LoadFromGlobal(globalCfg); err != nil {
+				return err
+			}
+			return app.RunOnce(context.Background(), cfg, nil, nil)
```

Comment:
I have thought about it again. Since there is already a context in the `Lightning` struct `RunOnce` method can always depend on the `app.ctx` instead of another ctx as parameter?

Reply:
In future usage, lightning may be `New` once and that instance `RunOnce` for many times, like a HTTP server mode. So a task context will help